tests/test_ujson.py

```diff
@@ -275,6 +275,53 @@ def test_encode_indent(test_input):
     assert output == json.dumps(obj, indent=4)
 
 
+def test_indent_types():
+    # Python json allows indent to be an int, str, or None.
+    # Check that ujson handles this as well.
+    data = [1, 2, 3]
+    output0a = ujson.encode(data)
+    output0b = ujson.encode(data, indent=None)
+    output0c = ujson.encode(data, indent="")
+    output0d = ujson.encode(data, indent=0)
+
+    output4a = ujson.encode(data, indent=4)
+    output4b = ujson.encode(data, indent="    ")
+
+    assert output0a == output0c
+    assert output0a == output0b
+    assert output0a == output0d
+
+    assert output4a == output4b
+
+    assert output0a != output4a
+
+
+def test_nonspace_indent():
+    # Nonspace inputs are allowed
+    assert ujson.encode([1], indent="900") == "[\n9001\n]"
+    assert ujson.encode([1], indent="\t") == "[\n\t1\n]"
+    assert ujson.encode([1], indent="\n  \t") == "[\n\n  \t1\n]"
+    indent = b"\xe6\x97\xa5\xd1\x88".decode("utf-8")
+    assert ujson.encode([1], indent=indent) == b"[\n\xe6\x97\xa5\xd1\x881\n]".decode(
+        "utf-8"
+    )
+
```

Comment:
An edge case that might be worth adding is surrogate escapes:

```suggestion
    assert ujson.encode([1], indent="\udc80") == "[\n\udc801\n]"

```

Reply:
Oh, and another idea: `indent = '\0'` (which would break if C string functions were involved for example)

Reply:
Interesting, so surrogate escapes don't work with the `PyUnicode` to `*char` conversion. Not entirely surprised by that. Perhaps there is no need to support it either since indentation can only consist of spaces, tabs, line feeds, and carriage returns per RFC 8259.

The NUL byte thing will also break for a different reason (output gets truncated on the first NUL byte on conversion back to a Python string), but probably no real reason to support it either.